### PR TITLE
docs(Terraform): add information about locking

### PR DESCRIPTION
**Pull request by afeld**
Closes https://github.com/cal-itp/benefits/issues/960.

Put that on a new Troubleshooting page, and moved the Monitoring and Logs information in there as well.

### CODE REVIEW

File: docs/configuration/environment-variables.md [hunk 1]
```diff
@@ -167,9 +166,8 @@ devcontainer, check the [`DJANGO_LOCAL_PORT`](#django_local_port).
 
     [Azure Monitor connection strings](https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string)
 
-Enables [log collection](../../deployment/infrastructure/#logs). Set the value in quotes, e.g. `APPLICATIONINSIGHTS_CONNECTION_STRING="InstrumentationKey=…"`.
+Enables [log collection](../../deployment/troubleshooting/#logs). Set the value in quotes, e.g. `APPLICATIONINSIGHTS_CONNECTION_STRING="InstrumentationKey=…"`.
 
 [app-service-config]: https://docs.microsoft.com/en-us/azure/app-service/configure-common?tabs=portal
 [benefits-secrets]: https://github.com/cal-itp/benefits-secrets
-[deployment]: ../deployment/README.md
```
Comment (afeld) [inline docs/configuration/environment-variables.md:174]: Was no longer used.

File: docs/deployment/infrastructure.md [hunk 2]
```diff
@@ -88,51 +88,6 @@ You'll see these referenced in Terraform as [data sources](https://developer.has
...
-```sh
-az webapp log tail --resource-group RG-CDT-PUB-VIP-CALITP-P-001 --name AS-CDT-PUB-VIP-CALITP-P-001 --slot dev 2>&1 | grep -v /healthcheck
-```
-
-`--slot dev` can be removed for production or changed for a different slot.
-
-### SCM
-
-https://as-cdt-pub-vip-calitp-p-001-dev.scm.azurewebsites.net/api/logs/docker
-
```
Comment (afeld) [inline docs/deployment/infrastructure.md:135]: Just moved these; no change.

File: docs/deployment/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+If Terraform commands fail (locally or in the Pipeline) due to an `Error acquiring the state lock`:
+
+1. Check the `Lock Info` for the `Created` timestamp. If it's **in the past ten minutes** or so, that probably means Terraform is still running elsewhere, and you should wait (stop here).
+1. **Are any [Pipeline runs](https://calenterprise.visualstudio.com/CDT.OET.CAL-ITP/_build?definitionId=828) stuck?** If so, cancel that build, and try re-running the Terraform command.
+1. **Do any engineers have a Terrafrom command running locally?** You'll need to ask them. For example: They may have started an `apply` and it's sitting waiting for them to [approve](https://developer.hashicorp.com/terraform/cli/commands/apply#automatic-plan-mode) it. They will need to (gracefully) exit for the lock to be relased.
+1. **If none of the steps above identified the source of the lock**, and especially if the `Created` time is more than ten minutes ago, that probably means the last Terraform command didn't release the lock. You'll need to grab the `ID` from the `Lock Info` output and [force unlock](https://developer.hashicorp.com/terraform/language/state/locking#force-unlock).
```
Comment (afeld) [inline docs/deployment/troubleshooting.md:57]: New stuff.

**Review by thekaveman — APPROVED** [review]
Thanks for the reorg as well!

File: docs/deployment/troubleshooting.md (hunk 3 above)
Comment (machikoyasuda) [reply docs/deployment/troubleshooting.md:57]: Could you add a link to this [issue](https://github.com/cal-itp/benefits/issues/960) that has helpful links to examples of Pull Requests with this error and the exact messages ?

Comment (afeld) [reply docs/deployment/troubleshooting.md:57]: The Benefits Azure DevOps Project is configured to only retains the Pipeline runs for 30 days, so unfortunately those won't be visible. I can link to the issue if you feel strongly; anything that I should have copied over but didn't?

**Review by machikoyasuda — APPROVED** [review]

**Pull request merged**